src/catalog/src/information_schema/views.rs

```diff
@@ -0,0 +1,288 @@
+        VIEWS
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.schema.clone()
+    }
+
+    fn to_stream(&self, request: ScanRequest) -> Result<SendableRecordBatchStream> {
+        let schema = self.schema.arrow_schema().clone();
+        let mut builder = self.builder();
+        let stream = Box::pin(DfRecordBatchStreamAdapter::new(
+            schema,
+            futures::stream::once(async move {
+                builder
+                    .make_views(Some(request))
+                    .await
+                    .map(|x| x.into_df_record_batch())
+                    .map_err(|err| {
+                        datafusion::error::DataFusionError::External(format!("{err:?}").into())
+                    })
+            }),
+        ));
+        Ok(Box::pin(
+            RecordBatchStreamAdapter::try_new(stream)
+                .map_err(BoxedError::new)
+                .context(InternalSnafu)?,
+        ))
+    }
+}
+
+/// Builds the `information_schema.VIEWS` table row by row
+///
+/// Columns are based on <https://dev.mysql.com/doc/refman/8.4/en/information-schema-views-table.html>
+struct InformationSchemaViewsBuilder {
+    schema: SchemaRef,
+    catalog_name: String,
+    catalog_manager: Weak<dyn CatalogManager>,
+
+    catalog_names: StringVectorBuilder,
+    schema_names: StringVectorBuilder,
+    table_names: StringVectorBuilder,
+    view_definitions: StringVectorBuilder,
+    check_options: StringVectorBuilder,
+    is_updatable: BooleanVectorBuilder,
+    definer: StringVectorBuilder,
+    security_type: StringVectorBuilder,
+    character_set_client: StringVectorBuilder,
+    collation_connection: StringVectorBuilder,
+}
+
+impl InformationSchemaViewsBuilder {
+    fn new(
+        schema: SchemaRef,
+        catalog_name: String,
+        catalog_manager: Weak<dyn CatalogManager>,
+    ) -> Self {
+        Self {
+            schema,
+            catalog_name,
+            catalog_manager,
+            catalog_names: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            schema_names: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            table_names: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            view_definitions: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            check_options: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            is_updatable: BooleanVectorBuilder::with_capacity(INIT_CAPACITY),
+            definer: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            security_type: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            character_set_client: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+            collation_connection: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+        }
+    }
+
+    /// Construct the `information_schema.views` virtual table
+    async fn make_views(&mut self, request: Option<ScanRequest>) -> Result<RecordBatch> {
+        let catalog_name = self.catalog_name.clone();
```

Comment:
Avoid unnecessary cloning; use `catalog_name` instead of `&catalog_name` since it's always referenced below. Clone only when essential:

```suggestion
        let catalog_name = &self.catalog_name;
```

Reply:
I think the clone here is necessary because its borrowed as a mutable reference with add_view() call